xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
> but we get one frame ahead. assume we have not called swapBuffers for a while. now the first call to swapBuffers will not block because of double buffers.

Yes, eglSwapBuffers will not block on the call after the 40 ms block, since it already received the frame sync from the compositor at that point. But it will block again right at the next call after that to resync with the compositor even if you have rendered faster. I assume on X/intel you could catch up because there are more buffers on the client side?

> if swapBuffers blocks, it makes a difference.

Exactly, that's why I asked: Take the time before or after swapBuffers?

> this code gets the time since last vblank. not sure if I get your question

Question was if that is somehow related to the problem at hand. Guess the answer means "no" ;-)